lib/daru/core/group_by.rb

```diff
@@ -35,9 +36,15 @@ def initialize context, names
         keys    = tuples.uniq.sort(&TUPLE_SORTER)
 
         keys.each do |key|
-          @groups[key] = all_indices_for(tuples, key)
+          indices = all_indices_for(tuples, key)
+          @groups[key] = indices
+          indices.each do |indice|
+            multi_index_tuples << key + [indice]
```

Comment:
Adding `key` and index value, to make tuple. 

```
multi_index_tuples

[["Jane", 1], ["Jane", 4], ["John", 0], ["John", 3], ["Mark", 2], ["Mark", 5]] 

```

groups

```
{["Jane"]=>[1, 4], ["John"]=>[0, 3], ["Mark"]=>[2, 5]}
```

keys are `[["Jane"], ["John"], ["Mark"]]`